### PR TITLE
WP.com API: Include Errors Listed for Broken Themes

### PR DESCRIPTION
**Pull request by Aurorum**
Needed for Automattic/wp-calypso#93132

## Proposed changes:

* Includes the errors of Broken Themes in the WP.com API. The aim is to include them in Calypso, but I think that I'd need some help with configuring this endpoint first.

## Jetpack product discussion

See Automattic/wp-calypso#93132.

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

As this is a change to the site endpoint, my ability to test is limited so please double-check the testing here:

- Install a theme on your site and break it

- You should see an error under the Themes section in WP-Admin like so.
Screenshot 2024-07-31 at 12 15 45

- With these changes, those errors should now be included in the endpoint like below. Confirm that is the case.

Screenshot 2024-07-31 at 12 12 43

Steps for breaking a theme (there's a lot of options but this is probably the easiest):

- Go to https://wordpress.com/theme/maywood
- Install and activate the theme
- Remove the Varia child theme on which Maywood (or any of the other Varia-based themes from WordPress.com) depend on

### CODE REVIEW

File: projects/plugins/jetpack/sal/class.json-api-site-base.php [hunk 1]
```diff
@@ -1057,6 +1057,29 @@ public function get_theme_slug() {
...
+		$themes_with_errors = wp_get_themes( array( 'errors' => true ) );
+		$theme_errors       = array();
+
+		foreach ( $themes_with_errors as $theme ) {
+			$errors = $theme->errors();
+
+			if ( ! empty( $errors->get_error_messages() ) ) {
+				$theme_errors[] = array(
+					'name'   => $theme->get( 'Name' ),
+					'errors' => $errors->get_error_messages(),
```
Comment (jeherve) [inline projects/plugins/jetpack/sal/class.json-api-site-base.php]: What do you think about sanitizing those values a tiny bit?

```suggestion
					'name'   => sanitize_title( $theme->get( 'Name' ) ),
					'errors' => (array) $errors->get_error_messages(),
```

File: projects/plugins/jetpack/sal/class.json-api-site-base.php [hunk 2]
```diff
@@ -1057,6 +1057,29 @@ public function get_theme_slug() {
...
+	 * @return array
+	 **/
+	public function get_theme_errors() {
+		$themes_with_errors = wp_get_themes( array( 'errors' => true ) );
+		$theme_errors       = array();
+
+		foreach ( $themes_with_errors as $theme ) {
+			$errors = $theme->errors();
+
+			if ( ! empty( $errors->get_error_messages() ) ) {
```
Comment (jeherve) [inline projects/plugins/jetpack/sal/class.json-api-site-base.php]: How about adding an additional check?

```suggestion
			if ( is_wp_error( $errors ) && ! empty( $errors->get_error_messages() ) ) {
```

File: projects/plugins/jetpack/sal/class.json-api-site-base.php [hunk 3]
```diff
@@ -1057,6 +1057,29 @@ public function get_theme_slug() {
 		return get_option( 'stylesheet' );
 	}
 
+	/**
+	 * Returns a list of errors for broken themes on the site.
+	 *
+	 * @return array
+	 **/
```
Comment (jeherve) [inline projects/plugins/jetpack/sal/class.json-api-site-base.php]:
```suggestion
	 */
```

File: projects/plugins/jetpack/sal/class.json-api-site-base.php (hunk 1 above)
Comment (Aurorum) [reply projects/plugins/jetpack/sal/class.json-api-site-base.php]: Thanks @jeherve! I agree with all your suggestions and should be added now. :)

**Review by jeherve — APPROVED** [review]

**Pull request merged**